src/wallet/wallet.cpp

```diff
@@ -4609,7 +4609,8 @@ std::shared_ptr<CWallet> CWallet::Create(interfaces::Chain& chain, interfaces::C
                     newHdChain.AddAccount();
                 } else {
                     if (gArgs.IsArgSet("-hdseed") && !IsHex(strSeed)) {
-                        walletInstance->WalletLogPrintf("%s -- Incorrect seed, generating a random mnemonic instead\n", __func__);
+                        error = strprintf(_("%s -- Incorrect seed, it should be a hex string"), __func__);
+                        return nullptr;
```

Comment:
I'm not sure I follow this... Why is it a problem to generate a new seed? (also what is bitcoins behavior in this way)

Reply:
In case if user specified argument `-hdseed` he wants to know a specific seed when creating wallet. Maybe it is his previous seed recovered from backup or paper wallet.

If user accidentally provided wrong seed (for example 58coded or mnemonic instead hex seed as expected), in this case instead returning failure it would be return succeed with random seed and a warning.

Firstly, the user can do not notice that warning and he may use wallet but he doesn't have this seed wrote down but user may think that he has. It can cause lost funds because seed of HD wallets is not backed up.

Secondly, even if the user will notice that wallet is created with wrong seed, it would be inconvenience because user now need to remove wallet with wrong seed and re-create it.

In one case that's annoying behavior, in other case that's dangerous behavior.

----------------
> also what is bitcoins behavior in this way

Bitcoin doesn't have this command line argument. But instead it has an rpc `sethdseed` and it throws error if key is invalid anyhow:

```
        CKey key = DecodeSecret(request.params[1].get_str());
        if (!key.IsValid()) {
            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid private key");
        }
```
here's it: https://github.com/bitcoin/bitcoin/blob/v27.0/src/wallet/rpc/wallet.cpp#L558-L571